src/main/java/wolox/training/web/controllers/BookController.java

```diff
@@ -0,0 +1,109 @@
+package wolox.training.web.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.web.dtos.BookCreationRequestDto;
+
+/**
+ * The {@link Book}s REST controller.
+ */
+@RestController
+@RequestMapping(value = "books", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+public class BookController {
+
+    /**
+     * The {@link BookRepository} used to read and write {@code Book} data.
+     */
+    private final BookRepository bookRepository;
+
+
+    /**
+     * Constructor.
+     *
+     * @param bookRepository The {@link BookRepository} used to read and write {@link Book} data.
+     */
+    @Autowired
+    public BookController(final BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```

Comment:
is there a reason to autowired it in the constructor and not just add the annotations on the attribute?

Reply:
There are several benefits. What I consider the main benefit is that is allows to use final variables.
Here is some info on the following sites: 
- https://stackoverflow.com/a/40620318/5373655
- https://www.vojtechruzicka.com/field-dependency-injection-considered-harmful/
- https://www.javacodegeeks.com/2015/01/dependency-injection-field-vs-constructor-vs-method.html
- https://blog.marcnuri.com/inyeccion-de-campos-desaconsejada-field-injection-not-recommended-spring-ioc/

Reply:
Excellent, did not know that.  👍